src/providers.rs

```diff
@@ -46,361 +39,219 @@ pub const LLAMA_ARBITRUM_ONE_HOSTNAME: &str = "arbitrum.llamarpc.com";
 pub const LLAMA_BASE_MAINNET_HOSTNAME: &str = "base.llamarpc.com";
 pub const LLAMA_OPTIMISM_MAINNET_HOSTNAME: &str = "optimism.llamarpc.com";
 
-// Limited API credentials for local testing.
-// Use `dfx canister call evm_rpc updateProvider ...` to pass your own keys.
-pub const ALCHEMY_ETH_MAINNET_CREDENTIAL: &str = "/v2/zBxaSBUMfuH8XnA-uLIWeXfCx1T8ItkM";
-pub const ALCHEMY_ETH_SEPOLIA_CREDENTIAL: &str = "/v2/Mbow19DWsfPXiTpdgvRu4HQq63iYycU-";
-pub const ALCHEMY_ARBITRUM_ONE_CREDENTIAL: &str = "/v2";
-pub const ALCHEMY_BASE_MAINNET_CREDENTIAL: &str = "/v2";
-pub const ALCHEMY_OPTIMISM_MAINNET_CREDENTIAL: &str = "/v2";
-pub const BLOCKPI_ETH_MAINNET_CREDENTIAL: &str = "/v1/rpc/0edc81e20be23ddff051f61a97bb457ec7284a58";
-pub const BLOCKPI_ETH_SEPOLIA_CREDENTIAL: &str = "/v1/rpc/1fe987fddded17db50862311720ff444991d4dab";
-pub const BLOCKPI_ARBITRUM_ONE_CREDENTIAL: &str =
-    "/v1/rpc/a8b89a41d2a341e32ee7aefcb20820a7cbb65f35";
-pub const BLOCKPI_BASE_MAINNET_CREDENTIAL: &str =
-    "/v1/rpc/bd458bf9f28ed45c77823814a937c812d2efd260";
-pub const BLOCKPI_OPTIMISM_MAINNET_CREDENTIAL: &str =
-    "/v1/rpc/d54bfe59299d56b0cbb8b3c69bd122f4ab5ac654";
-
-pub fn get_default_providers() -> Vec<RegisterProviderArgs> {
-    vec![
-        RegisterProviderArgs {
+thread_local! {
+    pub static PROVIDERS: Vec<Provider> = vec![
```

Comment:
I think here we could even simplify things further by having the list of providers not in stable memory but hard-coded in the wasm (i.e., as a `const`). For this we would need to tweak a bit the `Provider` struct. WDYT?

Reply:
The providers are no longer stored in stable memory (now just in a thread-local static value). 

I originally tried converting the providers to a constant, but this caused issues with using `format!()` to construct the URLs. After giving this another try, it appears that we can do this without any other issues.